### PR TITLE
Add AWS session-manager-plugin

### PR DESCRIPTION
**Pull request by manics**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/aws-session-manager-plugin/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  # Currently fails to build on Windows with Conda
+  skip: True  # [win] 
+
+requirements:
+  build:
+    - {{ compiler('go') }} 1.15
+    - go-licenses >=0.0.44
+    - make
+  run_constrained:
+    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx]
```
Comment (xhochy) [inline recipes/aws-session-manager-plugin/meta.yaml:23]: Not needed in this recipe
```suggestion
```

File: recipes/aws-session-manager-plugin/build.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -eu
+
+# https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html#environment-variables-set-during-the-build-process
+
+if [ -n "${MACOSX_DEPLOYMENT_TARGET:-}" ]; then
```
Comment (xhochy) [inline recipes/aws-session-manager-plugin/build.sh:6]:
```suggestion
if [[ "${target_platform}" == osx-* ]]; then
```

File: recipes/aws-session-manager-plugin/build.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -eu
+
+# https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html#environment-variables-set-during-the-build-process
```
Comment (xhochy) [inline recipes/aws-session-manager-plugin/build.sh:4]:
```suggestion
```

**Comment by manics** [comment]
@xhochy Thanks for reviewing and merging!

**Pull request merged**